File: 13768.Application/Services/ContactService.cs
using _13768.Application.Dtos;
using _13768.Application.Interfaces;
using _13768.Domain.Entities;

namespace _13768.Application.Services
{
    public class ContactService : IContactService
    {
        private readonly IContactRepository _contactRepository;

        public ContactService(IContactRepository contactRepository)
        {
              _contactRepository = contactRepository;
        }

        public void CreateContact(ContactDto contact)
        {
            var entity = new Contact
            {
                Name = contact.Name,
                Email = contact.Email,
                Notes = contact.Notes,
                Phone = contact.Phone,
                TeamId = contact.TeamId,
                Avatar = contact.Avatar,
                Address = contact.Address,
                JobTitle = contact.JobTitle,
                IsManager = contact.IsManager,
                ReportsTo = contact.ReportsTo,
                DateOfBirth = contact.DateOfBirth
            };

            _contactRepository.CreateContact(entity);

        }

        public void DeleteContact(int id)
        {
            var contact = _contactRepository.GetContact(id);

            if (contact != null)
            {
                _contactRepository.DeleteContact(contact);
            }
        }

        public Task<List<Contact>> GetAllAsync()
        {
            return _contactRepository.GetAllAsync();
        }

        public Task<List<Contact>> GetAllManagers()
        {
            return _contactRepository.GetAllManagersAsync();
        }

        public Contact? GetContact(int id)
        {
            return _contactRepository.GetContact(id);
        }

        public void UpdateContact(int id, ContactDto dto)
        {
            _contactRepository.UpdateContact(id, dto);
            
        }
    }
}

File: WAD.CW.13768/13768.AngularProject/13768.FrontEnd/src/app/services/contact.service.ts
import { Observable } from 'rxjs';
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { ContactCreateModel } from '../models/contact.create.model';
@Injectable({
  providedIn: 'root'
})
export class ContactService {

  constructor(private http: HttpClient) { }
  createUrl: string = "http://localhost:13768/api/Contact/Create";
  getAllUrl: string = "http://localhost:13768/api/Contact/GetAll";
  updateUrl: string = "http://localhost:13768/api/Contact/Update";
  deleteUrl: string = "http://localhost:13768/api/Contact/Delete";
  getByIdUrl: string = "http://localhost:13768/api/Contact/GetContactById";
  getAllManagersUrl: string = "http://localhost:13768/api/Contact/GetAllManagers";
  
  public createContact(contactDto: ContactCreateModel) : Observable<any>
  {
    const formData = new FormData();
    formData.append("Name", contactDto.name);
    formData.append("Email", contactDto.email);
    formData.append("Phone", contactDto.phone);
    formData.append("Notes", contactDto.notes);
    formData.append("TeamId", contactDto.teamId);
    formData.append("Avatar", contactDto.avatar);
    formData.append("Address", contactDto.address);
    formData.append("JobTitle", contactDto.jobTitle);
    formData.append("ReportsTo", contactDto.reportsTo ?? "");
    formData.append("DateOfBirth", contactDto.dateOfBirth.toString());
    formData.append("IsManager", contactDto.isManager == true ? "true" : "false");

    return this.http.post<any>(this.createUrl, formData);
  }

  public readContact(id: number): Observable<any>
  {
    return this.http.get<any>(`${this.getByIdUrl}/${id}`);
  }

  public readContacts(): Observable<any>
  {
    return this.http.get<any>(this.getAllUrl);
  }

  public updateContact(id: string, contactDto: ContactCreateModel): Observable<any>
  {
    const formData = new FormData();
    formData.append("Avatar", contactDto.avatar);
    formData.append("Email", contactDto.email);
    formData.append("Phone", contactDto.phone);
    formData.append("Notes", contactDto.notes);
    formData.append("Address", contactDto.address);
    formData.append("JobTitle", contactDto.jobTitle);
    formData.append("Name", contactDto.name);
    formData.append("DateOfBirth", contactDto.dateOfBirth.toString());
    formData.append("TeamId", contactDto.teamId);
    formData.append("ReportsTo", contactDto.reportsTo ?? "");
    formData.append("IsManager", contactDto.isManager == true ? "true" : "false");

    return this.http.post<any>(`${this.updateUrl}/${id}`, formData);
  }

  public deleteContact(id: string): Observable<any>
  {
    return this.http.delete<any>(`${this.deleteUrl}/${id}`);
  }

  public getAll() : Observable<any>
  {
    return this.http.get<any>(this.getAllUrl);
  }

  public getAllManagers() : Observable<any>
  {
    return this.http.get<any>(this.getAllManagersUrl);
  }
}
